specutils/io/default_loaders/tabular_fits.py

```diff
@@ -72,11 +72,37 @@ def tabular_fits_loader(file_name, column_mapping=None, **kwargs):
 
 
 @custom_writer("tabular-fits")
-def tabular_fits_writer(spectrum, file_name, **kwargs):
-    flux = spectrum.flux.value
-    disp = spectrum.spectral_axis.value
-    meta = spectrum.meta
-
-    tab = Table([disp, flux], names=("spectral_axis", "flux"), meta=meta)
-
-    tab.write(file_name, format="fits")
+def tabular_fits_writer(spectrum, file_name, update_header=False, **kwargs):
+    flux = spectrum.flux
+    disp = spectrum.spectral_axis
+    header = spectrum.meta.get('header', fits.header.Header()).copy()
+
+    if update_header:
+        hdr_types = (str, int, float, complex, bool,
+                     np.floating, np.integer, np.complexfloating, np.bool_)
+        header.update([k for k in spectrum.meta.items() if
+                       isinstance(k[1], hdr_types)])
+
+    # Strip header of FITS reserved keywords (strangely, loop does not work!)
+    if 'NAXIS' in header:
+        header.remove('NAXIS')
+    if 'NAXIS1' in header:
+        header.remove('NAXIS1')
+    if 'NAXIS2' in header:
+        header.remove('NAXIS2')
+
+    # Mapping of spectral_axis types to header TTYPE1
+    dispname = disp.unit.physical_type
+    if dispname == "length":
+        dispname = "wavelength"
+
+    columns = [disp, flux]
+    colnames = [dispname, "flux"]
+    # Include uncertainty - ToDo: uncertainty units?
```

Comment:
I think it's alright to not deal with uncertainty units because they should be assumed the same as the flux units.